### DISCUSSION TITLE
PDF Form - How do you bind the HTML Form and the PDF?

### DISCUSSION THREAD

**Discussion by Raahim** [stephanrauh/ngx-extended-pdf-viewer#1291] [Q&A] [answered]
Hi there,

I was looking at this example: https://pdfviewer.net/extended-pdf-viewer/forms

However, I do not fully understand how the HTML form is supoosed to be binded with the PDF form. Is there another basic example with complete source code that I could refer to?

I am fairly new to Angular so just trying to figure out from the above example has not been working for me. Any help will be greatly appreciated!

Thanks!

**Comment by stephanrauh** [comment, marked as answer]
Good point. Strictly speaking, it's not the HTML form that's linked with the Acroform (which is the form in the PDF file). There's just a two-way binding between the Acroform and a Json object in your Angular application.

My demo, in turn, also adds a two-way binding between the Angular HTML form and the same Acroform.

As a result, it looks like a two-way binding between the Acroform and the Angular HTML form, but in reality, it's a transitive link.

Does this shed some light?

**Reply by Raahim** [reply]
Oh that makes a lot more sense now! 

Thank you for the quick reply, I really apprecaite it!

PS. Great job on the package btw, it is very helpful! :)
